runtime/metricsview/executor.go

```diff
@@ -328,3 +332,288 @@ func (e *Executor) Export(ctx context.Context, qry *Query, executionTime *time.T
+		}
+		searchResult = append(searchResult, row)
+	}
+	if res.Err() != nil {
+		return nil, res.Err()
+	}
+	return searchResult, nil
+}
+
+func (e *Executor) executeSearchInDruid(ctx context.Context, olap drivers.OLAPStore, a *AST, search string) ([]SearchResult, error) {
+	if a.Root.FromSelect != nil {
+		// This means either the dimension uses an unnest or measure filters which are not directly supported by native search.
+		// This can be supported in future using query datasource in future if performance turns out to be a concern.
+		return nil, errDruidNativeSearchUnimplemented
+	}
+	var query map[string]interface{}
+	if a.Root.Where != nil {
+		// NOTE :: this does not work for measure filters.
+		// The query planner resolves them to joins instead of filters.
+		rows, err := olap.Execute(ctx, &drivers.Statement{
+			Query:            fmt.Sprintf("EXPLAIN PLAN FOR SELECT 1 FROM %s WHERE %s", *a.Root.FromTable, a.Root.Where.Expr),
+			Args:             a.Root.Where.Args,
+			DryRun:           false,
+			Priority:         0,
+			LongRunning:      false,
+			ExecutionTimeout: 0,
+		})
+		if err != nil {
+			return nil, err
+		}
+		defer rows.Close()
+
+		if !rows.Next() {
+			return nil, fmt.Errorf("failed to parse filter")
+		}
+
+		var (
+			planRaw string
+			resRaw  string
+			attrRaw string
+		)
+		err = rows.Scan(&planRaw, &resRaw, &attrRaw)
+		if err != nil {
+			return nil, err
+		}
+
+		var plan []druid.QueryPlan
+		err = json.Unmarshal([]byte(planRaw), &plan)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(plan) == 0 {
+			return nil, fmt.Errorf("failed to parse policy filter")
+		}
+		if plan[0].Query.Filter == nil {
+			// if we failed to parse a filter we return and run UNION query.
+			// this can happen when the row filter is complex
+			// TODO: iterate over this and integrate more parts like joins and subfilter in policy filter
+			return nil, errDruidNativeSearchUnimplemented
+		}
+		query = *plan[0].Query.Filter
+	}
+
+	// Build a native query
+	limit := 100
+	if a.Root.Limit != nil {
+		limit = int(*a.Root.Limit)
+	}
+	dims := make([]string, 0)
+	virtualCols := make([]druid.NativeVirtualColumns, 0)
+	for _, f := range a.Root.DimFields {
+		dim, err := a.lookupDimension(f.Name, true)
+		if err != nil {
+			return nil, err
+		}
+		// if the dimension is a expression we need a virtual column that can be scanned in SearchDimensions
+		if dim.Expression != "" {
+			virtualCols = append(virtualCols, druid.NativeVirtualColumns{
+				Type:       "expression",
+				Name:       fmt.Sprintf("%v_virtual_native", f.Name), // The name of the virtual column should not clash with actual column
+				Expression: dim.Expression,
+			})
+			dims = append(dims, fmt.Sprintf("%v_virtual_native", f.Name))
+		} else {
+			dims = append(dims, f.Name)
+		}
+	}
+	req := druid.NewNativeSearchQueryRequest(e.metricsView.Table, search, dims, virtualCols, limit, a.query.TimeRange.Start, a.query.TimeRange.End, query) // TODO: timestamps may be nil!
```

Comment:
Should this TODO be handled by returning `errDruidNativeSearchUnimplemented` if they are empty?

Reply:
I am not sure. I kept the existing todo as it is. I think this is not possible right now since UI sends time range in all cases. But yeah returning an error seems to be a good idea.